custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -194,6 +194,31 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         vm.selectedLocations = new Array(maxLevel);
     };
 
+    vm.userHaveAccessToAllLocations = function (locations) {
```

Comment:
I think all of these functions likely belong as helper functions inside the `locations.service` which this directive and the location filter can both call

Reply:
at a minimum it would be good to add a comment indicating where this was copy/pasted from and why it was done.

Reply:
sure.